synapse/types/__init__.py

```diff
@@ -695,6 +695,90 @@ async def to_string(self, store: "DataStore") -> str:
             return "s%d" % (self.stream,)
 
 
+@attr.s(frozen=True, slots=True, order=False)
+class MultiWriterStreamToken(AbstractMultiWriterStreamToken):
+    """A basic stream token class for streams that supports multiple writers."""
+
+    @classmethod
+    async def parse(cls, store: "DataStore", string: str) -> "MultiWriterStreamToken":
```

Comment:
Much of this matches `RoomStreamToken.parse` -- do we care?

We could maybe put this in the abstract and have that call `super()`...

Same goes for `to_string()` I think.

Reply:
Yeah, though its a bit fiddly with the topological part, so I figured it was easier to just duplicate the code for clarity :shrug: